File: 2206-divide-array-into-equal-pairs/2206-divide-array-into-equal-pairs.cpp
class Solution {
public:
    bool divideArray(vector<int>& nums) {
        unordered_map<int, int> map;
        for(int& x: nums ) map[x]++;
        for(auto& [f,s] : map)
        {
            if(s % 2 == 1) return false;
        }
        return true;
    }
};

File: 2206-divide-array-into-equal-pairs/2206-divide-array-into-equal-pairs.py
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        umap = {}
        for i in nums:
            if i not in umap:
                umap[i] = 1
            else:
                umap[i] += 1
        for k,v in umap.items():
            if v % 2:
                return False
        return True